File: backend/src/main/java/io/tingkai/money/entity/StockRecord.java
package io.tingkai.money.entity;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

import io.tingkai.money.constant.DatabaseConstants;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.Data;

@Entity
@Data
@Table(name = DatabaseConstants.TABLE_STOCK_RECORD, uniqueConstraints = { @UniqueConstraint(columnNames = { "code", "dealDate" }) })
public class StockRecord {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	protected UUID id;
	protected String code;
	protected LocalDateTime dealDate;
	protected BigDecimal dealShare;
	protected BigDecimal openPrice;
	protected BigDecimal highPrice;
	protected BigDecimal lowPrice;
	protected BigDecimal closePrice;
}

File: python/src/entity/StockRecord.py
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.schema import UniqueConstraint
import uuid

from entity.BaseEntity import BaseEntity, db

class StockRecord(BaseEntity):
    __tablename__ = 'stock_record'

    id = db.Column(UUID(as_uuid = True), primary_key = True, default = uuid.uuid4)
    code = db.Column(db.String, nullable = False)
    deal_date = db.Column(db.DateTime, nullable = False)
    deal_share = db.Column(db.Numeric, nullable = False)
    open_price = db.Column(db.Numeric, nullable = False)
    high_price = db.Column(db.Numeric, nullable = False)
    low_price = db.Column(db.Numeric, nullable = False)
    close_price = db.Column(db.Numeric, nullable = False)

    UniqueConstraint('code', 'deal_date', name='stock_record_code_deal_date_key')
